proxy-node-gateway/src/main/java/uk/gov/ida/notification/resources/EidasAuthnRequestResource.java

```diff
@@ -63,24 +66,27 @@ public View handlePostBinding(
             @FormParam(SamlFormMessageType.SAML_REQUEST) String encodedEidasAuthnRequest,
             @FormParam(RelayState.DEFAULT_ELEMENT_LOCAL_NAME) String eidasRelayState,
             @Session HttpSession session) {
-        return handleAuthnRequest(encodedEidasAuthnRequest, eidasRelayState, session);
+        return handleAuthnRequest(encodedEidasAuthnRequest, eidasRelayState, session.getId());
     }
 
-    private View handleAuthnRequest(String encodedEidasAuthnRequest, String eidasRelayState, HttpSession session) {
-        final EidasSamlParserResponse eidasSamlParserResponse = parseEidasRequest(encodedEidasAuthnRequest, session.getId());
-        AuthnRequestResponse vspResponse = generateHubRequestWithVsp(session.getId());
-        logAuthnRequestInformation(session, eidasSamlParserResponse, vspResponse);
-        setResponseDataInSession(session, eidasSamlParserResponse, vspResponse, eidasRelayState);
+    private View handleAuthnRequest(String encodedEidasAuthnRequest, String eidasRelayState, String sessionId) {
+        final EidasSamlParserResponse eidasSamlParserResponse = parseEidasRequest(encodedEidasAuthnRequest, sessionId);
+        AuthnRequestResponse vspResponse = generateHubRequestWithVsp(sessionId);
+        logAuthnRequestInformation(sessionId, eidasSamlParserResponse, vspResponse);
+        setResponseDataInSession(sessionId, eidasSamlParserResponse, vspResponse, eidasRelayState);
         return buildSamlFormView(vspResponse, eidasRelayState);
     }
 
-    private void setResponseDataInSession(HttpSession session, EidasSamlParserResponse eidasSamlParserResponse, AuthnRequestResponse vspResponse, String eidasRelayState) {
-        GatewaySessionData gatewaySessionData = new GatewaySessionData(
-            eidasSamlParserResponse,
-            vspResponse,
-            eidasRelayState
-        );
-        session.setAttribute(SESSION_KEY_SESSION_DATA, gatewaySessionData);
+    private void setResponseDataInSession(String sessionId, EidasSamlParserResponse eidasSamlParserResponse, AuthnRequestResponse vspResponse, String eidasRelayState) {
+        if (!sessionCacheManager.sessionExists(sessionId)) {
```

Comment:
Wouldn't this always be false? And if not, do we want to just carry on with the flow?